src/usb_moded-dbus.c

```diff
@@ -159,7 +184,10 @@ static DBusHandlerResult msg_handler(DBusConnection *const connection, DBusMessa
 		else
 		{
 			/* error checking is done when setting configuration */
-			if(!set_mode_setting(config))
+			int ret = set_mode_setting(config);
+			if (ret == SET_CONFIG_UPDATED)
+				usb_moded_send_config_signal(MODE_SETTING_ENTRY, MODE_SETTING_KEY, config);
```

Comment:
Issuing D-Bus signal from `set_config_setting` would introduce new dependency of usb_moded-config.c on dbus. To avoid introducing new dependencies, one could pass a callback, but it's easier to just return the status, IMO.